### ISSUE TITLE
resource/aws_ssm_parameter: Add data_type argument

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#13172]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Go SDK v1.30.21 includes support for SSM Parameters declaring a data type, which can be used for validation.

### Affected Resource

* aws_ssm_parameter

### Potential Terraform Configuration

```hcl
resource "aws_ssm_parameter" "example" {
  # ... other configuration ...
  data_type = "aws:ec2:image"
}
```

### References

* https://github.com/aws/aws-sdk-go/releases/tag/v1.30.21

**Comment by daniel-ciaglia** [comment]
Use case: https://docs.aws.amazon.com/systems-manager/latest/userguide/parameter-store-ec2-aliases.html

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.1.0 of the Terraform AWS Provider, likely next Thursday. Thanks to @jayolmos for the implementation. 👍
